sound/soc/sof/loader.c

```diff
@@ -873,6 +873,15 @@ int snd_sof_run_firmware(struct snd_sof_dev *sdev)
 		return ret;
 	}
 
+	mutex_lock(&sdev->cores_status_mutex);
+	/*
+	 * fw booted, Only the master DSP core (0) should be ON here,
+	 * update to reflect it.
+	 */
+	sdev->core_refs[0] = 1;
```

Comment:
this might need to be `sdev->core_refs[SOF_DSP_PRIMARY_CORE]` but if `SOF_DSP_PRIMARY_CORE` is ever `!= 0` would it still be an index in the array or just some random ID like 0x100?